app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/EditCoInsuredDestination.kt

```diff
@@ -172,21 +190,82 @@ private fun EditCoInsuredScreen(
             )
           }
         }
+
+        Column {
+          if (uiState.listState.priceInfo != null && uiState.listState.shouldShowPriceInfo()) {
+            Spacer(Modifier.height(8.dp))
+            PriceInfo(uiState.listState.priceInfo)
+            HedvigContainedButton(
+              text = stringResource(id = R.string.CONTRACT_ADD_COINSURED_CONFIRM_CHANGES),
+              onClick = {},
+              modifier = Modifier.padding(horizontal = 16.dp),
+            )
+          }
+
+          Spacer(Modifier.height(8.dp))
+          HedvigTextButton(
+            onClick = navigateUp,
+            text = stringResource(id = R.string.general_cancel_button),
+            modifier = Modifier.fillMaxWidth(),
+          )
+          Spacer(Modifier.height(16.dp))
+        }
       }
 
       EditCoInsuredState.Loading -> HedvigFullScreenCenterAlignedProgressDebounced()
     }
   }
 }
 
+@Composable
+private fun PriceInfo(priceInfo: EditCoInsuredState.Loaded.PriceInfo) {
+  val dateTimeFormatter = rememberHedvigDateTimeFormatter()
+
+  Column(
+    modifier = Modifier
+        .padding(16.dp)
+        .fillMaxWidth(),
+  ) {
+    Row(
+      horizontalArrangement = Arrangement.SpaceBetween,
+      modifier = Modifier.fillMaxWidth(),
+    ) {
+      Text(text = stringResource(id = R.string.CONTRACT_ADD_COINSURED_TOTAL))
+      Row(horizontalArrangement = Arrangement.End) {
+        Text(
+          text = stringResource(id = R.string.CHANGE_ADDRESS_PRICE_PER_MONTH_LABEL, priceInfo.previousPrice.toString()),
+          color = MaterialTheme.colorScheme.onSurfaceVariant,
+          style = LocalTextStyle.current.copy(textDecoration = TextDecoration.LineThrough),
+        )
+        Spacer(modifier = Modifier.width(8.dp))
+        Text(text = stringResource(id = R.string.CHANGE_ADDRESS_PRICE_PER_MONTH_LABEL, priceInfo.newPrice.toString()))
+      }
```

Comment:
Does this need the `HorizontalItemsWithMaximumSpaceTaken` treatment perhaps? What happens to the row on the right if the text on the left is long enough to take up the entire width?